File: BetterCalm/MSP.BetterCalm.Domain/Administrator.cs
using System;

namespace MSP.BetterCalm.Domain
{
    public class Administrator: User
    {
        public int AdministratorId { get; set; }
        public string Email { get; set; }
        public string Password { get; set; }
        public string Token { get; set; }

        protected bool Equals(Administrator other)
        {
            return Email == other.Email && 
                   Token == other.Token &&
                   Password == other.Password &&
                   Name == other.Name &&
                   LastName == other.LastName;
        }
        public override bool Equals(object obj)
        {
            if (ReferenceEquals(null, obj)) return false;
            if (ReferenceEquals(this, obj)) return true;
            if (obj.GetType() != this.GetType()) return false;
            return Equals((Administrator) obj);
        }

        public override int GetHashCode()
        {
            return HashCode.Combine(Email, Password);
        }

    }
}

File: BetterCalm/Frontend/src/app/models/Administrator.ts
export class Administrator {
  administratorId: number;
  name ?: string;
  lastName ?: string;
  email ?: string;
  password ?: string;
  token ?: string;

  constructor(
    administratorId: number,
    email ?: string,
    password ?: string,
    name ?: string,
    lastName ?: string,
    token ?: string,
  ) {
    this.administratorId = administratorId;
    this.email = email;
    this.password = password;
    this.name = name;
    this.lastName = lastName;
    this.token = token;
  }
}
